src/org/zaproxy/zap/extension/api/CoreAPI.java

```diff
@@ -429,6 +429,7 @@ public void run() {
 				Mode mode = Mode.valueOf(params.getString(PARAM_MODE).toLowerCase());
 		    	if (View.isInitialised()) {
 	    			View.getSingleton().getMainFrame().getMainToolbarPanel().setMode(mode);
+	    			View.getSingleton().getMainFrame().getMainMenuBar().setMode(mode);
```

Comment:
This has side effect of setting the mode twice (same when changing the mode through the Edit menu). The method Control.setMode(Mode) could be changed to ignore the mode (e.g. not notify of the change nor persist...) when already set.